### PR TITLE
chore: run prettier across repo

### PR DESCRIPTION
**Pull request by nschonni**
This touches a bunch of stuff because it looks like cleanup wasn't done yet. Here are the steps included. I could break this up a little, but the first step was the main cause of the changes.

- ran `npx prettier -w .` to autofix everything
- upgrade Prettier/ESLint prettier plugins to the latest versions
- wire `lint` target to run prettier and across all files
- add `fix` target to autofix ESLint/Prettier
- expand ESlint glob and fix issues in test files
- wire in ESLint prettier plugin that wasn't running
- update lint-staged targets to autofix if possible
- run lint on CI to prevent future issues

### Summary
:art::rotating_light::zap:

This pull request updates the code quality and formatting tools and settings for the project, and applies them to various files. It also fixes some minor code style and logic issues, and improves the readability and consistency of the code. The main tools used are `eslint` and `prettier`, and the main files affected are `.eslintrc.json`, `package.json`, `.github/workflows/pr.yml`, and `src/PolicyEngine.jsx`.

> _We are the masters of the code, we make it clean and neat_
> _We use the tools of `prettier` and `eslint` to defeat_
> _The chaos and the errors that lurk in every line_
> _We format and we lint, we are the code divine_

### Walkthrough
*  Add `prettier` plugin and configure `eslint` rules for code formatting and linting ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-6884918dc8291219be508e05e28965b958c734def85324f3b53858ea4702090fL8-R12),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-6884918dc8291219be508e05e28965b958c734def85324f3b53858ea4702090fL16-R24),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L15),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L56-R56),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L77-R85),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L105-R92))
*  Update `node-version` and `run` commands for GitHub Actions workflow ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-15c806aa509538190832852f439e9921a23bec2da81f95ed0e4bf13c14e5b160L28-R28),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-15c806aa509538190832852f439e9921a23bec2da81f95ed0e4bf13c14e5b160L36-R39))
*  Remove trailing spaces and commas from various files ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-15c806aa509538190832852f439e9921a23bec2da81f95ed0e4bf13c14e5b160L5-R5),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-f3fc934cf0d89bdf07de358896ff90f1202585df812cf606206d1830a9949811L3-R3),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-a5de3e5871ffcc383a2294845bd3df25d3eeff6c29ad46e3a396577c413bf357L2-R3),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-7fe0ab2d61296b5df4add971654584ba7ee8082cc5b79ffa3b60da0ed9f0b3ebL2-R4),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L2-R2))
*  Change `<!DOCTYPE html>` declaration and `link` elements to lowercase and add `/` for self-closing tags in `public/index.html` ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-c2cc24bc9001b11b6add48a4cd8f893d5d6c6e4d1bd254158bd14ab997f552cdL1-R1),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-c2cc24bc9001b11b6add48a4cd8f893d5d6c6e4d1bd254158bd14ab997f552cdL23-R28))
*  Add comma after `user-scalable=yes` attribute in `public/index.html` ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-c2cc24bc9001b11b6add48a4cd8f893d5d6c6e4d1bd254158bd14ab997f552cdL45-R48))
*  Remove unused imports from test files ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-ea82f6e985b4405746178eb385dfa12dba2ec298d08ea4e73718a83dc0f5eee3L1-R41),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-2f831bf8fe2ee1817acb1ff0b7d594d844d58ca96bee5f6a0639b9c7c0571918L5-R64),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-e217baee6f19e2e78342dc24d5d7a60120b6ef45e9466740f9d10fbec0af551cL5-R39))
*  Add commas to `MemoryRouter` elements in `src/__tests__/PolicyEngine.test.js` ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-1c2b80474596287b2950fe137106bc41085323dbcb252bf16922c48420f3a94fL11-R11),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-1c2b80474596287b2950fe137106bc41085323dbcb252bf16922c48420f3a94fL25-R25))
*  Add trailing commas to function parameters and arguments in `src/api/call.js`, `src/api/parameters.js` and `src/api/variables.js` ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-21944af492c6c9ee9158a5bc0a089aef67b9e83c6b6a9c1cce3dae2df054f259L27-R27),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-21944af492c6c9ee9158a5bc0a089aef67b9e83c6b6a9c1cce3dae2df054f259L79-R79),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-21944af492c6c9ee9158a5bc0a089aef67b9e83c6b6a9c1cce3dae2df054f259L85-R86),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-134c46ff4d3763aad565f9a0043f21e7375545cac7c4636e6b342c81f155753bL7-R7),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-134c46ff4d3763aad565f9a0043f21e7375545cac7c4636e6b342c81f155753bL38-R38),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-134c46ff4d3763aad565f9a0043f21e7375545cac7c4636e6b342c81f155753bL49-R49),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL207-R207),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL219-R219),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL228-R228),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL345-R345),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL413-R413),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL436-R436),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL446-R446),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL481-R481),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL494-R494),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL507-R507),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL513-R513),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-b5951a72ed1475216756f45c9a54ea34fb15cce384284eb48b32a30c5c0bf58bL532-R532))
*  Remove extra whitespace from destructuring assignments and attributes in `src/controls/InputField.jsx` and `src/controls/NavigationButton.jsx` ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-10323c70aaa8c36a4d474a2c79df4d291dc1a09fbf1e17b0c4ddae37404abd29L7-R7),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-10323c70aaa8c36a4d474a2c79df4d291dc1a09fbf1e17b0c4ddae37404abd29L71-R70),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-fc3284b09378d8360c8bcfe2e7e44d6cbe6c4e806becdf3ace25facd8bb619a1L7-R38))
*  Add comma to `whileHover` object and format `style` attribute and `props` object in `src/layout/BlogPostHolder.jsx` ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-8a958a370170ded33a0dc9f3090eb77d7f25540b0e0c3fe2f2707a042bcdd127L8-R19),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-8a958a370170ded33a0dc9f3090eb77d7f25540b0e0c3fe2f2707a042bcdd127L38-R48),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-8a958a370170ded33a0dc9f3090eb77d7f25540b0e0c3fe2f2707a042bcdd127L57-R68),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-8a958a370170ded33a0dc9f3090eb77d7f25540b0e0c3fe2f2707a042bcdd127L108-R118))
*  Change `gtag` function call to use camelCase keys and format `style` attribute and `margin` properties in `src/PolicyEngine.jsx` ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-8888a99b581bbcfa5b66bffe03f7de6872d52ef3deacccfcef6ce92af21b27c5L54-R60),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-8888a99b581bbcfa5b66bffe03f7de6872d52ef3deacccfcef6ce92af21b27c5L69-R68),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-8888a99b581bbcfa5b66bffe03f7de6872d52ef3deacccfcef6ce92af21b27c5L75-R117),[link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-8888a99b581bbcfa5b66bffe03f7de6872d52ef3deacccfcef6ce92af21b27c5L139-R157))
*  Format `element` attribute in `src/PolicyEngineCountry.jsx` ([link](https://github.com/PolicyEngine/policyengine-app/pull/658/files?diff=unified&w=0#diff-ba8020f07e0e8342911e79bf3646db148f6ffafd8c080eaee9390c2b690285d8L202-R207))

### CODE REVIEW

**Comment by nschonni** [comment]
@nikhilwoodruff I'm not sure what the CI failure is, that isn't showing up loccally. Let me know if yo want me to split this up , or if you wan to slowly land this lolaclly, on your own

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -25,14 +25,14 @@ jobs:
...
     steps:
       - uses: actions/checkout@v3
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v3
         with:
+          cache: npm
           node-version: ${{ matrix.node-version }}
-        # https://docs.npmjs.com/cli/v9/commands/npm-ci
-      - run: npm install --force
+      - run: npm run lint
```
Comment (nschonni) [inline .github/workflows/pr.yml:36]:
```suggestion
     - run: npm ci
      - run: npm run lint
```

**Comment by nikhilwoodruff** [comment]
Thanks @nschonni - taking a look at this now. If we can get the CI passing I think this would be great to merge in here!

**Comment by nschonni** [comment]
CI is good again. I mixed up the setup-node and the setup-ruby behaviour. The ruby one automagically does the install when you turn on caching, but the Node one doesn't. Now that I added back the `npm ci` it's happy again

**Review by nikhilwoodruff — APPROVED** [review]

**Pull request merged**